client/client.js

```diff
@@ -62,71 +63,97 @@ Client.create = function (message) {
 // Instance methods
 
 // Persist subscriptions and presences when not already persisted in memory
-Client.prototype.storeData = function (message) {
-  var subscriptions = this.subscriptions,
-      presences = this.presences,
-      changed = false,
+Client.prototype.storeData = function (messageIn) {
+  var subCount = Object.keys(this.subscriptions).length,
+      presCount = Object.keys(this.presences).length,
+      logNow = false;
+
+  // Persist the message data, according to type
+  switch(messageIn.op) {
+    case 'unsubscribe':
+    case 'sync':
+    case 'subscribe':
+      logNow = this._storeDataSubscriptions(messageIn, subCount);
+      break;
+
+    case 'set':
+      logNow = this._storeDataPresences(messageIn);
+      break;
+  }
+
+  if (logNow) {
+    log.info('#storeData: client_id: ' + this.id +
+              '; subscription count: ' + subCount +
+              '; presence count: ' + presCount);
+  }
+
+  return true;
+};
+
+Client.prototype._storeDataSubscriptions = function (messageIn, subCount) {
+  var message = _cloneForStorage(messageIn),
+      to = message.to,
+      subscriptionHashname = this.key + '_subs',
+      logNow = false,
       lookupMessage;
 
   // Persist the message data, according to type
   switch(message.op) {
     case 'unsubscribe':
-      if (subscriptions[message.to]) {
-        delete subscriptions[message.to];
-        changed = true;
+      if (this.subscriptions[to]) {
+        delete this.subscriptions[to];
+        Core.Persistence.expire(subscriptionHashname, Client.getDataTTL());
+        Core.Persistence.deleteHash(subscriptionHashname, to);
+        logNow = 0 === subCount % LOG_WINDOW_SIZE;
       }
       break;
 
     case 'sync':
     case 'subscribe':
-      lookupMessage = subscriptions[message.to];
+      lookupMessage = this.subscriptions[to];
       if (!lookupMessage ||
             (lookupMessage.op != 'sync' && !_.isEqual(lookupMessage, message))) {
-        subscriptions[message.to] = message;
-        changed = true;
+        this.subscriptions[to] = message;
+        Core.Persistence.expire(subscriptionHashname, Client.getDataTTL());
+        Core.Persistence.persistHash(subscriptionHashname, to, message);
+        logNow = 0 === subCount % LOG_WINDOW_SIZE;
       }
       break;
+  }
+
+  return logNow;
+};
+
+Client.prototype._storeDataPresences = function (messageIn, presCount) {
+  var message = _cloneForStorage(messageIn),
+      to = message.to,
+      presenceHashname = this.key + '_pres',
```

Comment:
I would prefer the keys to be clearer. `+ '/subscriptions'` and `+ '/presences'` will work better I think.